internal/internal_decision_state_machine.go

```diff
@@ -440,7 +472,7 @@ func (d *commandStateMachineBase) handleCancelFailedEvent() {
 
 func (d *commandStateMachineBase) handleCanceledEvent() {
 	switch d.state {
-	case commandStateCancellationCommandSent:
+	case commandStateCancellationCommandSent, commandStateCanceledAfterInitiated, commandStateCanceledBeforeSent:
```

Comment:
How can state machine in the state `commandStateCanceledBeforeSent` when handling `CanceledEvent`.  Server cannot generate canceled event on its own.  The command has to be sent for the server to generate the event.

Reply:
So the idea here was these new `cancelTimerStateMachine` and `cancelActivityStateMachine` machines are the ones that generate the command, and so the original machines then handle their events. That said, though, probably it means these states should not be in this base handler, but explicitly in the activity/timer handlers.

Reply:
Indeed at least here the `commandStateCanceledBeforeSent` can be removed without regressing anything so this one is better now.